mobly/controllers/android_device.py

```diff
@@ -408,6 +408,8 @@ def start_services(self, skip_sl4a=False):
         except:
             self.log.exception("Failed to start adb logcat!")
             raise
+        if skip_sl4a is None:
+            skip_sl4a=getattr(ad, KEY_SKIP_SL4A, False)
```

Comment:
I see.
Now I realize that it is a bit odd to affect the function's behavior by setting attribute on an object...